src/main/scala/swiss/dasch/domain/BulkIngestService.scala

```diff
@@ -38,22 +38,26 @@ final case class BulkIngestService(
       _         <- semaphoresPerProject.put(key, semaphore)
     } yield semaphore).commit
 
-  private def acquireWithTimeout(sem: TSemaphore): UIO[Unit] =
-    sem.acquire.commit.timeout(Duration.fromMillis(400)).as(())
+  private def acquireWithTimeout(sem: TSemaphore): IO[Unit, Unit] =
+    sem.acquire.commit.timeout(Duration.fromMillis(400)).some.mapError(_ => ())
```

Comment:
`.mapError(_ => foo)` can be replaced with `.orElseFail(foo)`

Reply:
Great!